test/integration/src/test/java/io/pravega/test/integration/RestoreBackUpDataRecoveryTest.java

```diff
@@ -395,36 +464,41 @@ public void testDurableDataLogFail() throws Exception {
         // Start a debug segment container corresponding to the given container Id and put it in the Hashmap with the Id.
         Map<Integer, DebugStreamSegmentContainer> debugStreamSegmentContainerMap = new HashMap<>();
 
-        // Create a debug segment container instance using a new dataLog and old storage.
-        DebugStreamSegmentContainerTests.MetadataCleanupContainer debugStreamSegmentContainer = new
-                DebugStreamSegmentContainerTests.MetadataCleanupContainer(CONTAINER_ID, CONTAINER_CONFIG, localDurableLogFactory,
-                context.readIndexFactory, context.attributeIndexFactory, context.writerFactory, this.storageFactory,
-                context.getDefaultExtensions(), executorService());
+        // Create a debug segment container instances using a new dataLog and old storage.
+        for (int containerId = 0; containerId < containerCount; containerId++) {
+            DebugStreamSegmentContainerTests.MetadataCleanupContainer debugStreamSegmentContainer = new
+                    DebugStreamSegmentContainerTests.MetadataCleanupContainer(containerId, CONTAINER_CONFIG, localDurableLogFactory,
+                    context.readIndexFactory, context.attributeIndexFactory, context.writerFactory, this.storageFactory,
+                    context.getDefaultExtensions(), executorService());
 
-        Services.startAsync(debugStreamSegmentContainer, executorService()).get(TIMEOUT.toMillis(), TimeUnit.MILLISECONDS);
-        debugStreamSegmentContainerMap.put(CONTAINER_ID, debugStreamSegmentContainer);
+            Services.startAsync(debugStreamSegmentContainer, executorService()).get(TIMEOUT.toMillis(), TimeUnit.MILLISECONDS);
+            debugStreamSegmentContainerMap.put(containerId, debugStreamSegmentContainer);
 
-        // Delete container metadata segment and attributes index segment corresponding to the container Id from the long term storage
-        ContainerRecoveryUtils.deleteMetadataAndAttributeSegments(storage, CONTAINER_ID).get(TIMEOUT.toMillis(), TimeUnit.MILLISECONDS);
+            // Delete container metadata segment and attributes index segment corresponding to the container Id from the long term storage
+            ContainerRecoveryUtils.deleteMetadataAndAttributeSegments(storage, containerId).get(TIMEOUT.toMillis(), TimeUnit.MILLISECONDS);
+        }
```

Comment:
try to see if we can break it into small methods

Reply:
Broke it into few.